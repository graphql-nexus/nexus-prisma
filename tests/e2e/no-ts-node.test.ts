import dindist from 'dindist'
import 'ts-replace-all'
import { createPrismaSchema } from '../__helpers__/testers'
import { setupTestProject } from '../__helpers__/testProject'

it('when project does not have ts-node installed nexus-prisma generator still generates if there are no TS generator config files present', async () => {
  const testProject = setupTestProject({
    files: {
      packageJson: {
        scripts: {
          build: 'prisma generate',
        },
        dependencies: {
          '@prisma/client': '2.30',
          graphql: '15.5.1',
          nexus: '1.1.0',
          prisma: '2.30',
        },
      },
      other: [
        {
          filePath: `prisma/schema.prisma`,
          content: createPrismaSchema({
            content: dindist`
							model Foo {
								id  String  @id
							}
						`,
          }),
        },
      ],
    },
  })

  const result = await testProject.runOrThrowNpmScript(`build`)

  expect(result.stdout.replaceAll(/\d+ms/g, 'XXXms')).toMatchSnapshot()
})
