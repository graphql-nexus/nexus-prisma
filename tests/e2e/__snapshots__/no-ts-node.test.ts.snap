// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when project does not have ts-node installed nexus-prisma generator still generates if there are no TS generator config files present 1`] = `
"Prisma schema loaded from prisma/schema.prisma
You can now start using Nexus Prisma in your code. Reference: https://pris.ly/d/nexus-prisma

✔ Generated Prisma Client (2.30.0) to ./node_modules/@prisma/client in XXXms

✔ Generated Nexus Prisma to ./.yalc/nexus-prisma/dist/runtime in XXXms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
\`\`\`
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
\`\`\`"
`;
